e2e/internal/e2e/keyexec.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func getImportScript(kpath string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+set psk "e2etests"
+
+spawn singularity key import %s
+
+expect "Enter your old password : "
+send "${psk}\r"
+
+expect "Enter a new password for this key : "
+send "${psk}\r"
+
+expect "Retype your passphrase : "
+send "${psk}\r"
+
+expect eof
+`, kpath)
+}
+
+func getExportScript(num int, kpath, armor, psk string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+spawn singularity key export --secret %s %s
+
+expect "Enter # of signing key to use : "
+send "%v\r"
+
+expect "Enter key passphrase : "
+send "%s\r"
+
+expect eof
+`, armor, kpath, num, psk)
+}
+
+var backupSypgp = filepath.Join(HomeDir(), ".singularity/sypgp/secret-keyring-backup")
+
+// PullDefaultPublicKey will pull the public Sylabs Admin key
+func PullDefaultPublicKey(t *testing.T) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "pull", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+// RemoveDefaultPublicKey will pull the public Sylabs Admin key
+func RemoveDefaultPublicKey(t *testing.T) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "remove", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
```

Comment:
First, the command will not fail if theres no key. Second, if there is no key when this function is called, theres a another problem. So I dont think its necessary to check if the key exist first...

Reply:
Uhm...

in order to help with automation, the command _should_ return with a non-zero exit code if the key to be removed does not exist. If that's currently not the case, an issue should be opened for that, and a test should be added, so that we the code changes, the test catches it.

As to the "other problem"... what might that be?

My original point is that in a test you should make sure that all the preconditions for the test are met. If I see this test without any other context, I would expect that there's something making sure this particular key exists. I don't see that here, which makes the test code harder to understand.

Reply:
>First, the command will not fail if theres no key.

Nevermined, it does fail if theres no key.

Any ideas on how to check if this key is in the keyring?